test/client_tests/client_test.dart

```diff
@@ -173,7 +173,7 @@ void main() {
 
     await harness.runFailureTest(
       clientCall: harness.client.unary(dummyValue),
-      expectedException: GrpcError.unimplemented('No responses received'),
+      expectedException: GrpcError.unknown('missing :status header'),
```

Comment:
I'm not quite sure if this is the way to go, because the former exception seems to fit better?

Reply:
This can be fixed by sending `:status` as part of the trailers. gRPC protocol assumes that OnlyTrailers responses contain both trailers and headers. 

I am going to push a commit on top of your work fixing this.